libr/util/str_trim.c

```diff
@@ -108,6 +108,59 @@ R_API void r_str_trim_head(char *str) {
 	}
 }
 
+/* remove spaces between args of the string except if args are surrounding
+ * by quotes.
+ * the string is changed in place */
+R_API void r_str_trim_args(char *str) {
+	const char *quotes = "\"'";
+	const char *isquote;
+	int idx, _b = 0, q = 0;
+	ut8 *b = (ut8*)&_b;
+	char *ws, *ch = str;
+
+	ws = (char *)r_str_trim_head_wp (ch);
+	while (*ws) {
+		ch = (char *)r_str_trim_head_ro (ws);
+		if (*ch) {
+			isquote = strchr (quotes, *ch);
+			if (isquote) {
+				idx = (int)(size_t)(isquote - quotes);
+				_b = R_BIT_TOGGLE (b, idx);
+				q = 1;
+			}
+			memmove (ws + 1, ch + q, strlen (ch + q) + 1);
+		} else {
+			break;
+		}
+		if (q == 1) {
+			q = 0;
+			for (; *ch; ch++) {
+				if (*ch == '\\') {
+					ch++;
+					continue;
+				}
+				isquote = strchr (quotes, *ch);
+				if (isquote) {
+					idx = (int)(size_t)(isquote - quotes);
+					_b = R_BIT_TOGGLE (b, idx);
+					if (!_b) {
+						break;
+					} else {
+						_b = R_BIT_TOGGLE (b, idx);
+					}
+				}
+			}
+			if (!*ch) {
+				break;
+			}
+			memmove (ch, ch + 1, strlen (ch));
```

Comment:
same. try to avoid calling strlen in loops when you can predict string length

Reply:
Done.